cmd/pro/add/cluster.go

```diff
@@ -0,0 +1,312 @@
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type ClusterCmd struct {
+	Log log.Logger
+	*proflags.GlobalFlags
+
+	Namespace        string
+	ServiceAccount   string
+	DisplayName      string
+	KubeContext      string
+	Insecure         bool
+	Wait             bool
+	HelmChartPath    string
+	HelmChartVersion string
+	HelmSet          []string
+	HelmValues       []string
+	Host             string
+}
+
+// NewClusterCmd creates a new command
+func NewClusterCmd(globalFlags *proflags.GlobalFlags) *cobra.Command {
+	cmd := &ClusterCmd{
+		GlobalFlags: globalFlags,
+		Log:         log.GetInstance(),
+	}
+
+	c := &cobra.Command{
+		Use:   "cluster",
+		Short: "add current cluster to DevPod Pro",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cobraCmd *cobra.Command, args []string) error {
+			devPodConfig, err := config.LoadConfig(cmd.Context, "")
+			if err != nil {
+				return err
+			}
+
+			if cmd.Host == "" {
+				proInstances, err := workspace.ListProInstances(devPodConfig, cmd.Log)
+				if err != nil {
+					return fmt.Errorf("list pro instances: %w", err)
+				}
+				options := []string{}
+				for _, pro := range proInstances {
+					options = append(options, pro.Host)
+				}
+				host, err := cmd.Log.Question(&survey.QuestionOptions{
+					Question:     "Select Pro instance to connect your cluster to",
+					Options:      options,
+					DefaultValue: options[0],
+				})
+				if err != nil {
+					return fmt.Errorf("select pro instance: %w", err)
+				}
+				cmd.Host = host
+			}
+
+			return cmd.Run(cobraCmd.Context(), args, devPodConfig)
+		},
+	}
+
+	c.Flags().StringVar(&cmd.Namespace, "namespace", "loft", "The namespace to generate the service account in. The namespace will be created if it does not exist")
+	c.Flags().StringVar(&cmd.ServiceAccount, "service-account", "loft-admin", "The service account name to create")
+	c.Flags().StringVar(&cmd.DisplayName, "display-name", "", "The display name to show in the UI for this cluster")
+	c.Flags().BoolVar(&cmd.Wait, "wait", false, "If true, will wait until the cluster is initialized")
+	c.Flags().BoolVar(&cmd.Insecure, "insecure", false, "If true, deploys the agent in insecure mode")
+	c.Flags().StringVar(&cmd.HelmChartVersion, "helm-chart-version", "", "The agent chart version to deploy")
+	c.Flags().StringVar(&cmd.HelmChartPath, "helm-chart-path", "", "The agent chart to deploy")
+	c.Flags().StringArrayVar(&cmd.HelmSet, "helm-set", []string{}, "Extra helm values for the agent chart")
+	c.Flags().StringArrayVar(&cmd.HelmValues, "helm-values", []string{}, "Extra helm values for the agent chart")
+	c.Flags().StringVar(&cmd.KubeContext, "kube-context", "", "The kube context to use for installation")
+	c.Flags().StringVar(&cmd.Host, "host", "", "The pro instance to use")
+
+	return c
+}
+
+func (cmd *ClusterCmd) Run(ctx context.Context, args []string, devPodConfig *config.Config) error {
```

Comment:
this function is pretty big, maybe it could only call some middle level functions like `getClients` `prepareHelmCommand` `installAgentIntoCluster` or something like this? This way flow and what's actually happening is clear when reading top level function

Reply:
I'd like to use it for a bit before breaking it up. This is literally what we do with vCluster and I'm not sure yet if we need to adjust it or not